Runtime/Device/ButtonRing/AdxIO4ButtonRing.cs

```diff
@@ -120,10 +122,10 @@ change this function to safe and copy the bytes this way.
         public override void ReadData(ReadOnlySpan<byte> data)
         {
             // Check if the state has changed
-            if (ByteArraysEqual(_currentState, data))
-            {
-                return;
-            }
+            //if (ByteArraysEqual(_currentState, data))

```

Comment:
why remove this check? if current state and data are equal there is no reason to process the packet? Removing this should make things less efficient.